### DISCUSSION TITLE
modcharts

### DISCUSSION THREAD

**Discussion by Ibexlol** [KadeArchive/Kade-Engine#1588] [Q&A]
how do steps work in modcharts?

**Comment by glowsoony** [comment]
steps in mod chart work like this: a example of  my mod chart


function start (song)
	print("Song: " .. song .. " @ " .. bpm .. " donwscroll: " .. downscroll)
end


function update (elapsed)
    local currentBeat = (songPos / 1000)*(bpm/190)
        for i=0,7 do
        setActorY(defaultStrum0Y + 18 * math.cos((currentBeat + i*0.95) * math.pi), i)
        end
end

function beatHit (beat)

end

function stepHit (step)
    if curStep == 120 or curStep == 121 or curStep == 122 or curStep == 123 or curStep == 124 or curStep == 125 or curStep == 126 or curStep == 248 or curStep == 290 or curStep == 344 or curStep == 376 or curStep == 486 or curStep == 600 or curStep == 660 or curStep == 806 or curStep == 808 or curStep == 810 or curStep == 952 or curStep == 1000 or curStep == 1100 then
        for i = 0, 7 do
            tweenPosXAngle(i, _G['defaultStrum'..i..'X'],getActorAngle(i) + 180, 0.20, 'setDefault')
        end
    end

**Reply by Ibexlol** [reply]
thank you

**Reply by glowsoony** [reply]
no problem :)
